### PR TITLE
feat: Sentry integration

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This is a fairly standard Sentry setup.  It might help diagnose https://github.com/freeCodeCamp/coderadio-client/issues/80

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -12,6 +13,15 @@ const SUB = new NchanSubscriber(
   "wss://coderadio-admin.freecodecamp.org/api/live/nowplaying/coderadio"
 );
 
+SUB.on("error", function(err, errDesc) {
+  Sentry.addBreadcrumb({
+    message: "NchanSubscriber error: " + errDesc
+  });
+  // I'm assuming captureException is appropriate here, I'm not sure what type
+  // the first argument has.
+  Sentry.captureException(err);
```
Comment (nhcarrigan) [inline src/components/App.js:22]: I dove into the declarations files for the module and the `exception` parameter is typed as `any`. 

```js
/**
 * Captures an exception event and sends it to Sentry.
 *
 * @param exception An exception-like object.
 * @returns The generated eventId.
 */
export declare function captureException(exception: any, captureContext?: CaptureContext): string;
```

File: src/components/App.js [hunk 2]
```diff
@@ -371,7 +385,10 @@ export default class App extends React.Component {
...
     );
 
-  onPlayerError = () => {
+  onPlayerError = err => {
+    // NOTE: this might be a bit noisy, so we should monitor it before promoting
+    // coderadio
+    Sentry.captureException(err);
     /*
      * This error handler works as follows:
      * - When the player cannot play the url:
```
Comment (nhcarrigan) [inline src/components/App.js:394]: I plan on leaving this running locally for a while so if this handler starts exploding I'll let you know. 🙂

**Review by nhcarrigan — COMMENTED** [review]
I pulled this down locally, set up Sentry, and got my first breadcrumb. Seems to be a pretty streamlined process.

I did get one error(?) in Sentry that seems to be specific to running this locally:
image

But this shouldn't affect prod as far as I know.

File: src/components/App.js (hunk 1 above)
Comment (ojeytonwilliams) [reply src/components/App.js:22]: Thanks Nick. We can maybe get clearer output, but this should be okay.

**Comment by ojeytonwilliams** [comment]
I saw those errors too and came to the same conclusion.  Presumably it's trying to fetch something from our local machines.

**Comment by ojeytonwilliams** [comment]
Hey @ahmadabdolsaheb we've got the Sentry project set up.  When you've got a chance could you give it a review and merge if you're happy?

File: src/components/App.js (hunk 2 above)
Comment (ahmadabdolsaheb) [reply src/components/App.js:390]: so the funny thing with that error is that the player does not error out :)

Comment (nhcarrigan) [reply src/components/App.js:390]: Right so it wouldn't catch here, but it should catch in the `NchanSubscriber` handler right?

https://github.com/freeCodeCamp/coderadio-client/blob/f39b74e03b1e6184fc9ebfc56b5db0ac1a1a3e72/src/components/App.js#L16-L22

**Comment by ojeytonwilliams** [comment]
FYI I've deployed a build without the DSN since we were getting rate-limited by Sentry (thanks for spotting that @nhcarrigan).  I'll investigate later and re-deploy once I'm happy it's not going to explode.

**Pull request merged**